yapio.c

```diff
@@ -725,6 +732,33 @@ yapio_parse_test_recipe(const char *recipe_str)
             i += yapio_test_recipe_param_to_ull(&recipe_str[i + 1], &tmp);
             ytg->ytg_num_ranks = (int)tmp;
             break;
+        case 'P':
+            if (recipe_str[i + 1] != '.')
+            {
+                printf("Percentage should start by '.'\n");
+                yapio_print_help(YAPIO_EXIT_ERR);
+            }
+            /* find where percentage stops */
+            int end = i + 2;
+            while (end < recipe_str_len)
+            {
+                if (!isdigit(recipe_str[end]))
+                    break;
+                end++;
+            }
+            unsigned long int percent = 0;
+            percent = strtoull(&recipe_str[i + 2], NULL, 10);
+
+            /* count number of skipped IOs */
+            ytc->ytc_sparse_io = percent * ytg->ytg_num_blks_per_rank;
+            int j;
+            for (j = i + 2; j < end; j++)
+                ytc->ytc_sparse_io = ytc->ytc_sparse_io/10;
```

Comment:
I fail to see the value of that code. But i might be wrong

 You already have the length of percentage, i'll bet (end - i +2) so why not do that in a one liner ? i would say something line the following ?
 ytc->ytc_sparse_io = (percent * ytg->ytg_num_blks_per_rank) / ((end- i +2) * 10) ; 

You'll do less operation thus less approximation error which would be dramatique if you want to choose something really close to 1 like .999999 for instance

Furthermore you are missing one division by 10 in the current setup of you loop.

Reply:
I think it's more like *10^len(percent)*. I don't see how to avoid the successive divisions.

If P.0100 and num_blks_per_rank = 100 we should skip 1% of 100 IOs so that's 1 IO.
We get (variable) percent = 100 and len(percent) = 4
therefore: IO skipped = 100 *  100 / 10^4 = 1